validation/validate_results.R

```diff
@@ -0,0 +1,160 @@
+    table_diffs <- difference_between_lists(ref_tables, test_tables)
+
+    if (length(table_diffs$reference_only) > 0) {
+        logging::loginfo(sprintf("%s - The following tables exist only in the REFERENCE database:\n%s",
+                                 event_name, table_diffs$reference_only))
+    }
+    if (length(table_diffs$test_only) > 0) {
+        logging::loginfo(sprintf("%s - The following tables exist only in the TEST database:\n%s",
+                                 event_name, table_diffs$test_only))
+    }
+
+    # compare columns in tables
+    for (table in table_diffs$common) {
+        ref_columns <- RSQLite::dbGetQuery(ref_db_con, sprintf(table_columns_query, table))$names
+        test_columns <- RSQLite::dbGetQuery(test_db_con, sprintf(table_columns_query, table))$names
+        column_diffs <- difference_between_lists(ref_columns, test_columns)
+
+        if (length(column_diffs$reference_only) > 0) {
+            logging::loginfo(sprintf("%s - Table %s: the following columns exist only in the REFERENCE data:\n%s",
+                                     event_name, table, column_diffs$reference_only))
+        }
+        if (length(column_diffs$test_only) > 0) {
+            logging::loginfo(sprintf("%s - Table %s: the following columns exist only in the TEST data:\n%s",
+                                     event_name,  table, column_diffs$test_only))
+        }
+
+        # compare volume of data in columns
+        ref_table_length <- RSQLite::dbGetQuery(ref_db_con, sprintf(table_length_query, table))
+        test_table_length <- RSQLite::dbGetQuery(test_db_con, sprintf(table_length_query, table))
+
+        if (ref_table_length$length[[1]] != test_table_length$length[[1]]) {
+            logging::loginfo(sprintf(
+                "%s - Table %s has differing lengths in reference and test data\nref: %s; test: %s",
+                event_name, table, ref_table_length$length[[1]], test_table_length$length[[1]]
+                ))
+        }
+ 
+        # compare values in table
+        if (compare_values) {
+            ref_data <- RSQLite::dbGetQuery(ref_db_con, sprintf(table_all_data_query, table))
+            test_data <- RSQLite::dbGetQuery(test_db_con, sprintf(table_all_data_query, table))
+            diffs <- all.equal(ref_data[column_diffs$common], ref_data[column_diffs$common])
+            if (!isTRUE(diffs)){
+                logging::loginfo(
+                    sprintf(
+                        "%s - Differences found in values:\n%s",
+                        event_name, toString(diffs)
+                    )
+                )
+            }
+        }
+    }
+}
+
+
+compare_dfs <- function(reference, test, event_name=NA){
+    # 1. check for new columns
+    ref_columns <- names(reference)
+    test_columns <- names(test)
+    column_diff <- difference_between_lists(ref_columns, test_columns)
+
+    if (length(column_diff$reference_only) > 0) {
+        logging::loginfo(
```

Comment:
It would be useful to also see which dfs are being compared when looking at the log

Reply:
I'm now printing which CSV/dfs are being testing when this function is triggered